lib/tizen_project.dart

```diff
@@ -72,6 +70,10 @@ class TizenProject extends FlutterProjectPlatform {
   @override
   bool existsSync() => editableDirectory.existsSync();
 
+  bool get isDotnet => projectFile.path.endsWith('.csproj');
+
+  bool get isApplication => manifestFile.existsSync();
```

Comment:
`TizenProject` is an implementation of the `FlutterProjectPlatform` class, so it is expected to be a representation of **a platform specific sub-project of a Flutter project** but not **a plugin project**. There is a class called `TizenPlugin` that exactly meets your purpose.

In your design,
- `TizenProject` may represent: either an app project (Tizen sub-project) or a dotnet plugin project.
- `TizenPlugin` may represent: either C++/dotnet/Dart plugin project.

I think it would be better to
- Move the `_updateDotnetUserProjectFile` function outside the `TizenProject` class (still you can put it in the same file) and make it public.
- Create a method in the `TizenPlugin` class that has a similar role to `TizenProject`'s `ensureReadyForPlatformSpecificTooling`.
- Call the method on each dotnet plugin from `injectTizenPlugins`.

This means that you may need to partially undo https://github.com/flutter-tizen/flutter-tizen/pull/341 and https://github.com/flutter-tizen/flutter-tizen/pull/337.

Reply:
I think that the `TizenProject` refers to a project located in the `tizen` sub-directory under the flutter project regardless of the project type. For Android, The `AndroidProject` is responsible for not only apps but also modules and plugins. And `TizenPlugin` is more about storing information obtained from `pubspec.yaml` than representing a plugin project.

However, your suggestions will help simplify the code. I will try to make `_updateDotnetUserProjectFile` public.

Reply:
Oh, I just forgot about the fact that a `FlutterProject` can be either app, module, or plugin. You can ignore my comment.

How about replacing `isApplication` with `isPlugin` similarly to `AndroidProject.isPlugin`?

Reply:
ok, `isApplication` can be removed for now. `FlutterProject.isPlugin` can be used instead.